src/tools/idlc/src/descriptor.c

```diff
@@ -1268,8 +1272,8 @@ emit_declarator(
       return ret;
 
     /* in case there is no revisit required (array has simple element type) we have
-       to close the mutable member immediately, otherwise close it when revisiting */
-    if (mutable_aggr_type_member && (!(ret & IDL_VISIT_REVISIT) || revisit)) {
+       to close the mutable member immediately */
+    if (mutable_aggr_type_member) {
```

Comment:
Although this change is not incorrect, I think a better fix would be to modify the check on line 1271 so that it only returns in case of an error:
```C
if ((ret = emit_array(pstate, revisit, path, node, user_data)) < 0) 
  return ret
```

Then the condition on line 1276 for closing the mutable member remains unchanged: the member is immediately closed in case the emit function does not set the `IDL_VISIT_REVISIT` bit, or it is closed in case of revisiting.

Reply:
Fixing the `if` on line 1271 as I suggested above reveals a bug on line 1280, this line should be: `return ret;`